### DISCUSSION TITLE
PostgreSQL Composite Type in SQLALChemy

### DISCUSSION THREAD

**Discussion by GFuhr** [sqlalchemy/sqlalchemy#10101] [Usage Questions]
I'm new to SQLAlchemy so sorry for my very basic question and for my english.

I would like to use SQLALchemy 2.0+ psycopg3 to interact with a PostgreSQL Database using Composite Type. What I would like to be able to do using SQLAlchemy is to reproduce this SQL Script : 
```
"""
CREATE TYPE complex AS (
    r       double precision,
    i       double precision
);

CREATE TABLE forecast (
    probes      complex,
    ref     integer
);

INSERT INTO forecast VALUES (ROW( 42.42, 1.99), 1000);
"""
```
I looked at the Composite Type in SQLAlchemy but it doesn't correspond to the postgresql composite type from what I understood. I would prefer to have a way to do it without having to use sqlalchemy-utils to avoid the psycopg2 package dependency.

thank you in advance,
G.

**Comment by CaselIT** [comment]
Hi,

This is basically the same question as this discussion from yesterday https://github.com/sqlalchemy/sqlalchemy/discussions/10095

Reply is there isn't much support at the moment for them. You may be able to insert the data from sqlalchemy though if you define the type and the table in sql directly

**Comment by GFuhr** [comment]
Hi,
thanks for this answer even if it wasn't the one I was hoping for :sweat_smile:

So based on your answer, if I define the type and table in SQL, after I should be able to use this kind of example : https://docs.sqlalchemy.org/en/20/orm/composites.html ?

G.

**Reply by CaselIT** [reply]
no, that does basically the opposite of what you want: uses multiple table columns on the db and maps them on a single python object in the model.

I think the best option is probably use sqlalchemy-utils if you must use composite types in the db
